pkg/segment/segment.go

```diff
@@ -211,3 +218,59 @@ func IsTelemetryEnabled(cfg *preference.PreferenceInfo) bool {
 	}
 	return false
 }
+
+type contextKey struct{}
```

Comment:
It could be interesting to have this code in a separate package. 

You are using a generic storage (map[string]interface{}), but you are restricting the possibilites to add values to this storage by exporting only `SetComponentType`, and not `set`.

- If you want to have a generic storage, you shouldn't have a `SetComponentType` but only a `Set` (exported),
- If you want to restrict the possibilites of values to store, you could continue to have `SetComponentType` and have a struct like:

```
type Properties struct {
        componentType string
}
```

Reply:
I do want to restrict the possibility of values stored in the context. I tried what you have suggested, i.e. having a Properties struct, but it seemed to get complicated, mainly for 2 reasons -
1. We are currently using maps, which are easier to deal with mainly because they allow easy iteration.
2. We are sending this data to segment.io along with some other information, if we use struct, we'll need to convert them to maps so that we can iterate and all information there is, or we will need to explicitly add every information.

Perhaps I am missing something, it might help if you can elaborate on your suggestion.

>It could be interesting to have this code in a separate package.

It makes sense to move to a separate package if we will use it somewhere other than here, else I can move the code to a different file within the same package for better readability.

Reply:
> I do want to restrict the possibility of values stored in the context. I tried what you have suggested, i.e. having a Properties struct, but it seemed to get complicated, mainly for 2 reasons -
> 
> 1. We are currently using maps, which are easier to deal with mainly because they allow easy iteration.
> 2. We are sending this data to segment.io along with some other information, if we use struct, we'll need to convert them to maps so that we can iterate and all information there is, or we will need to explicitly add every information.
> 
> Perhaps I am missing something, it might help if you can elaborate on your suggestion.
> 

OK, storing the data in the desired form is a good idea. Let's keep it as is.

> > It could be interesting to have this code in a separate package.
> 
> It makes sense to move to a separate package if we will use it somewhere other than here, else I can move the code to a different file within the same package for better readability.

If you want to restrict the possible values and prevent the future developers to be tempted to use the `set` function directly, it would be better to place these functions in a separate package, so the `set` function is unaccessible. In a different file of  the same package, it will remain accessible.

Reply:
Done. I moved the context to another package within the segment package since I think it is more relevant to it and I don't expect people to use it for other packages.